### ISSUE TITLE
NamedTuples defined via Python 3.6 variable annotations should allow methods

### ISSUE THREAD

**Issue by roganov** [python/typing#352]
Following code should work as expected

```python
In [1]: from typing import NamedTuple

In [2]: class X(NamedTuple):
   ...:     x: int
   ...:     def double(self):
   ...:         return 2 * x
   ...:     

In [3]: X(1).double()
```

At the moment it throws `AttributeError` as there is no `double` method on the constructed namedtuple.

The workaround is to define methods on subclass and set `__slots__ = ()`. But unless I'm missing something, there is no reason why methods shouldn't work as expected.

**Comment by ilevkivskyi** [comment]
Initially I implemented this syntax to mimic the old ``namedtuple`` API. But now I looks like there will be no problem allowing arbitrary methods (also in mypy). 

@gvanrossum what do you think?

**Comment by gvanrossum** [comment]
I think it's fine.

**Comment by JelleZijlstra** [comment]
Thanks! I'll work on implementing this in mypy as soon as I finish implementing async generators and will add a doc patch to CPython once this is merged.

**Comment by ilevkivskyi** [comment]
@roganov Please check that you are happy with PR #362.

**Comment by roganov** [comment]
@ilevkivskyi Seems fine, thank you!

**Comment by ambv** [comment]
Well, still needs typing to be synchronized with the CPython repo.

**Comment by gvanrossum** [comment]
Done. (Normally we don't keep issues open once the fix is committed to master.)

**Comment by ilevkivskyi** [comment]
I have made a small follow-up PR #364 
It adjusts the behaviour in case of a collision between a method name and ``NamedTuple`` field or special attribute. Now such method is silently ignored, but I think it is better to be "loud" here and raise an ``AttributeError`` (to avoid hiding obscure bugs):
```python
class X(NamedTuple):
    x: int
    def _fields(self):  # This will be an error
        ...
```
@gvanrossum If you are not against this, then I will merge it tomorrow.
